lb.go

```diff
@@ -59,14 +59,18 @@ func (l *lb) BalanceLoad(req grpclb.LoadBalancer_BalanceLoadServer) error {
 	}
 }
 
-func convertServerList(l ServerList, offset int) []*grpclb.Server {
-	var servers []*grpclb.Server
-
+func convertServerList(l ServerList, max int) []*grpclb.Server {
 	n := len(l)
+	if max > n || max == 0 {
+		max = n
+	}
+	servers := make([]*grpclb.Server, max)
 
-	for i := 0; i < n; i++ {
-		server := l[(i+offset)%n]
-		servers = append(servers, convertServer(server))
+	rand.Seed(time.Now().UnixNano())
+	for i := 0; i < max; i++ {
+		j := rand.Intn(n-i) + i
+		l[i], l[j] = l[j], l[i]
+		servers[i] = convertServer(l[i])
 	}
```

Comment:
Is it some sort of shuffling algorithm? If so I would consider to move it to separate function